p2p/switch.go

```diff
@@ -370,6 +370,12 @@ func (sw *Switch) stopAndRemovePeer(peer Peer, reason interface{}) {
 	// https://github.com/tendermint/tendermint/issues/3338
 	if sw.peers.Remove(peer) {
 		sw.metrics.Peers.Add(float64(-1))
+	} else {
+		// Removing the peer has failed so we set a flag to mark that a removal was attempted
+		// This is done to avoid the corner case of adding a peer when this error happens before the
+		// AddPeer routine has finished
```

Comment:
Can you just extend this to inform the reader what setting that flag means i.e. that once it's set the node won't add that peer again

Reply:
Done.